tests/monitor.rs

```diff
@@ -4,82 +4,87 @@
 pub mod common;
 
 use assert_cmd::cargo::cargo_bin;
-use codec::Decode;
-use common::{init_logger, run_polkadot_node, KillChildOnDrop};
-use sp_storage::StorageChangeSet;
-use staking_miner::{
-	opt::Chain,
-	prelude::{runtime, sp_core::Bytes, Hash, SubxtClient},
+use common::{
+	init_logger, run_staking_miner_playground, spawn_cli_output_threads, test_submit_solution,
+	KillChildOnDrop, Target,
 };
+use regex::Regex;
+use staking_miner::opt::Chain;
 use std::{
 	process,
 	time::{Duration, Instant},
 };
-use subxt::rpc::rpc_params;
-use tokio::time::timeout;
-
-const MAX_DURATION_FOR_SUBMIT_SOLUTION: Duration = Duration::from_secs(60 * 15);
 
 #[tokio::test]
 async fn submit_monitor_works_basic() {
 	init_logger();
-	test_submit_solution(Chain::Polkadot).await;
-	test_submit_solution(Chain::Kusama).await;
-	test_submit_solution(Chain::Westend).await;
+	test_submit_solution(Target::PolkadotNode(Chain::Polkadot)).await;
+	test_submit_solution(Target::PolkadotNode(Chain::Kusama)).await;
+	test_submit_solution(Target::PolkadotNode(Chain::Westend)).await;
+	// It's easier to debug this by just connection a staking miner to
+	// staking-miner-playground by `./target/debug/staking-miner --uri ws://localhost:9944 monitor --seed-or-path //Alice --listen finalized seq-phragmen`
+	test_submit_solution(Target::StakingMinerPlayground).await;
 }
 
-async fn test_submit_solution(chain: Chain) {
-	use runtime::runtime_types::pallet_election_provider_multi_phase::{
-		ElectionCompute, ReadySolution,
-	};
-
-	let (_drop, ws_url) = run_polkadot_node(chain);
-
-	let _miner = KillChildOnDrop(
+#[tokio::test]
+async fn default_trimming_works() {
+	init_logger();
+	let (_drop, ws_url) = run_staking_miner_playground();
+	let miner = KillChildOnDrop(
 		process::Command::new(cargo_bin(env!("CARGO_PKG_NAME")))
 			.stdout(process::Stdio::piped())
 			.stderr(process::Stdio::piped())
+			.env("RUST_LOG", "runtime=debug,staking-miner=debug")
 			.args(&["--uri", &ws_url, "monitor", "--seed-or-path", "//Alice", "seq-phragmen"])
 			.spawn()
 			.unwrap(),
 	);
 
-	let api = SubxtClient::from_url(&ws_url).await.unwrap();
+	assert!(has_trimming_output(miner).await);
+}
+
+// Helper that parses the CLI output to find logging outputs based on the following:
+//
+// i) DEBUG runtime::election-provider: 🗳 from 934 assignments, truncating to 1501 for weight, removing 0
+// ii) DEBUG runtime::election-provider: 🗳 from 931 assignments, truncating to 755 for length, removing 176
+//
+// Thus, the only way to ensure that trimming actually works.
+async fn has_trimming_output(mut miner: KillChildOnDrop) -> bool {
```

Comment:
I think we can remove this ugly stuff, if the solution is correct and accepted by the chain it's sufficient as we ensure that trimming always occurs on the staking-miner-playground

Reply:
hmmm, not convinced. If you mis-configure this such that the trimming functions do nothing, how can we identify that.

Reply:
yeah true, we need to have this and ensure it's not trimming output != 0 for weight and length